scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/ParquetAvroIO.scala

```diff
@@ -201,6 +201,16 @@ object ParquetAvroIO {
       job.getConfiguration.setClass("key.class", classOf[Void], classOf[Void])
       job.getConfiguration.setClass("value.class", avroClass, avroClass)
 
+      // Needed to make GenericRecord read by parquet-avro work with Beam's
+      // org.apache.beam.sdk.coders.AvroCoder.
+      if (ScioUtil.classOf[A] == classOf[GenericRecord]) {
```

Comment:
Maybe we should use `classOf[GenericRecord].isAssignableFrom(ScioUtil.classOf[A])` ? I've seen some pipelines that are typed `GenericData.Record` -  which is the [default implementation of GenericRecord](https://avro.apache.org/docs/1.8.1/api/java/org/apache/avro/generic/GenericData.Record.html).

Reply:
actually, nvm, because that would include `SpecificRecord`s too.

Reply:
yeah, it wouldn't work, but I think you concern is valid

Reply:
after thinking about it more, I think it is ok to support only GenericRecord bc `GenericData.Record` is a default implementation and there is no guarantee it is returned by `parquet-avro` anyway + since we are reading the interface is more important because a user don't need to construct an instance of a record.